pkg/sql/alter_table.go

```diff
@@ -953,6 +953,10 @@ func applyColumnMutation(
 		); err != nil {
 			return err
 		}
+		if col.HasNullDefault() {
+			// `SET DEFAULT NULL` means a nil default expression.
+			col.ColumnDesc().DefaultExpr = nil
+		}
```

Comment:
Dumb question perhaps but shouldn't this logic rather be in `updateNonComputedColExpr`? It feels like this should also apply to `ON UPDATE` expressions.